includes/base/element-base.php

```diff
@@ -395,6 +395,8 @@ public function add_link_attributes( $element, array $url_control, $overwrite =
 			$attributes = array_merge( $attributes, Utils::parse_custom_attributes( $url_control['custom_attributes'] ) );
 		}
 
+		$attributes = apply_filters( 'elementor/element/link-attributes', $attributes, $url_control );
```

Comment:
Seems like this is not the right place, this method is a helper method to add attributes to a link, I think it should be called when the link renders.